packages/core/src/features/css-class.ts

```diff
@@ -0,0 +1,198 @@
+import { getOriginDefinition } from '../helpers/resolve';
+import { namespaceEscape } from '../helpers/escape';
+import { flattenFunctionalSelector, convertToClass, stringifySelector } from '../helpers/selector';
+import type { StylableMeta } from '../stylable-meta';
+import type { ScopeContext } from '../stylable-transformer';
+import { valueMapping } from '../stylable-value-parsers';
+import { validateRuleStateDefinition } from '../helpers/custom-state';
+import { ignoreDeprecationWarn } from '../helpers/deprecation';
+import type {
+    ImmutableClass,
+    Class,
+    FunctionalSelector,
+    ImmutableSelectorNode,
+} from '@tokey/css-selector-parser';
+import type * as postcss from 'postcss';
+
+export interface ClassSymbol extends StylableDirectives {
+    _kind: 'class';
+    name: string;
+    alias?: ImportSymbol;
+    scoped?: string; // ToDo: check if in use
+}
+
+const dataKey = plugableRecord.key<Record<string, ClassSymbol>>();
+
+export const diagnostics = {
+    INVALID_FUNCTIONAL_SELECTOR: generalDiagnostics.INVALID_FUNCTIONAL_SELECTOR,
+    UNSCOPED_CLASS(name: string) {
+        return `unscoped class "${name}" will affect all elements of the same type in the document`;
+    },
+};
+
+// HOOKS
+
+export const hooks = createFeature({
+    analyzeInit({ data }) {
+        plugableRecord.set(data, dataKey, {});
+    },
+    analyzeSelectorNode<AST extends ImmutableClass>(
+        meta: StylableMeta,
+        node: AST,
+        rule: postcss.Rule
+    ): void {
+        if (node.nodes) {
+            // error on functional class
+            meta.diagnostics.error(
+                rule,
+                diagnostics.INVALID_FUNCTIONAL_SELECTOR(`.` + node.value, `class`),
+                { word: stringifySelector(node) }
+            );
+        }
+        addClass(meta, node.value, rule);
+    },
+    transformSelectorNode<AST extends Class>(context: Required<ScopeContext>, node: AST): void {
+        const { originMeta, resolver } = context;
+        const resolved = context.metaParts.class[node.value] || [
+            // used to namespace classes from js mixins since js mixins
+            // are scoped in the context of the mixed-in stylesheet
+            // which might not have a definition for the mixed-in class
+            { _kind: 'css', meta: originMeta, symbol: { _kind: 'class', name: node.value } },
+        ];
+        context.setCurrentAnchor({ name: node.value, type: 'class', resolved });
+        const { symbol, meta } = getOriginDefinition(resolved);
+        if (context.originMeta === meta && symbol[valueMapping.states]) {
+            // ToDo: refactor out to transformer validation phase
+            validateRuleStateDefinition(context.rule, meta, resolver, originMeta.diagnostics);
+        }
+        namespaceClass(meta, symbol, node, originMeta);
+    },
+});
+
+// API
+
+export function getClass({ data }: StylableMeta, name: string): ClassSymbol | undefined {
```

Comment:
unify with `getSymbols` API

Reply:
unified features symbol api

- `getClass`, `getType`, `getPart`, `getSymbol` -> `get`
- `getSymbols` => `getAll`

I kept `addSymbol`, `addClass`, `addType` (and the non existing `addPart`) separate for now as they have different signatures and are easy to change later if we decide to do so, because they are only used internally in the core process.